refactor: Rename `PHCheckbox` to `LemonCheckbox` and rework it

Twixes opened the pull request:
## Problem

A key part of the funnel table I'm working on is checkboxes. I noticed this checkbox component of ours `PHCheckbox`, but was a bit surprised by the name, as we don't use the `PH` prefix anywhere else.
Then I realized it's not actually an `input` OR a `label` – it was implemented as plain `div`s all the way, which is really bad for accessibility.

## Changes

This replaces `PHCheckbox` with `LemonCheckbox`. This new checkbox component has much better accessibility (e.g. `label` that's attached automatically so that it's always clickable too) and sparks a bit more joy (smooth check/uncheck transition):

![Bloop](https://user-images.githubusercontent.com/4550621/160156789-9485aedb-be80-41d8-bd63-27370d15f149.gif)

## How did you test this code?

Made the Storybook story shown above.

Twixes committed:
refactor: Rename `PHCheckbox` to `LemonCheckbox` and rework it

clarkus reviewed (approved):
Looks good to me. There is a special case for checkboxes that are used to toggle the value of other checkboxes. They can represent a mixed set of values via an "indeterminate" state. It's usually used in a table header to facilitate bulk actions, which we don't do yet, but we will likely need in the feature. It renders as a horizontal bar centered in the bounds of the input. It's a work in progress, but I have some of this documented at https://www.figma.com/file/Y9G24U4r04nEjIDGIEGuKI/PostHog-Design-System-One?node-id=1812%3A517.

Twixes commented:
Yep, that's supported too, you can play with this in this PR's Storybook (Details link next to Storybook Publish at the bottom of this PR's checks).

alexkim205 commented on frontend/src/lib/components/LemonCheckbox/LemonCheckbox.scss:
```diff
@@ -0,0 +1,52 @@
+    background: var(--bg-light);
+    border: 1px solid var(--border-dark);
+    border-radius: var(--radius);
+    --fill: var(--primary);
```
just curious, what does --fill do here?

Twixes replied:
Oops, this was leftover from a previous approach where `color` was set with a CSS var, that turned out to be cumbersome. Removed.

alexkim205 commented on frontend/src/lib/components/LemonCheckbox/LemonCheckbox.tsx:
```diff
@@ -0,0 +1,78 @@
+        setLocalChecked(checked)
+    }, [checked])
+
+    id = useMemo(() => id || `lemon-checkbox-${Math.floor(Math.random() * 1000000)}`, [id])
```
Any reason we're not creating a new variable id immutably here?

Twixes replied:
What do you mean by "immutably"?

mariusandra replied:
I guess Alex means replacing:

```tsx
function Component() {
    const id = useMemo(() => rawId || `lemon-checkbox-${Math.floor(Math.random() * 1000000)}`, [rawId])
}
```

with

```tsx
let i = 0
function Component() {
    const id = useMemo(() => rawId || `lemon-checkbox-${i++}`, [rawId])
}
```

Although small, there's a legit chance someone somewhere will eventually have two checkboxes that get their values synced because of the same random ID, leading to some weirdness.

Twixes replied:
I see, so basically "sequential". Made it so.

alexkim205 reviewed (changes requested):
Nice, thanks for bringing another component over to Lemon land. Just a few suggestions inline and some considerations below:

1. Checkboxes in InsightTable seem to be disabled and unclickable.

<img width="244" alt="Screen Shot 2022-03-25 at 10 38 36 AM" src="https://user-images.githubusercontent.com/13460330/160173254-a5655902-517d-4174-8259-1fb8a0e7cc61.png">
 
2. Should cursor be a pointer here when clicking on the checkbox labels?

https://user-images.githubusercontent.com/13460330/160173825-ad5fb734-274a-471b-b3e3-77021451e193.mov

3. Minor: Changing from indeterminate state to unchecked breaks seems broken. Seems like it starts the animation from a checked state rather from the indeterminate state.

https://user-images.githubusercontent.com/13460330/160174160-cca8534f-dc25-498c-803a-560daa3dbb57.mov

4. Might be out of scope (existing bug perhaps) but checkbox in column shouldn't be indeterminate when all row checkboxes are checked in funnel step view.

<img width="344" alt="Screen Shot 2022-03-25 at 10 46 02 AM" src="https://user-images.githubusercontent.com/13460330/160174235-14f32df7-debc-484b-ae5f-1e5683f83723.png">

Twixes committed:
Fix issues

Twixes commented:
Thanks for taking a look!
1. Hm, these checkboxes being disabled seems to be the case in `master` too.
2. I actually googled this question today (whether it's better for checkboxes to have `cursor: pointer`) as I've been seeing some opinions on the web that `pointer` is overused on buttons. My conclusion was basically: `pointer` is perfectly fine as an additional tool to help users find what's clickable, but that's not the default, and hover/action styles may in fact help this in a visually nicer way. So not really sure which is better, but I think I'll leave as is for now.
3. Great catch. Fixed.
4. Fixed too.

Twixes committed:
Add `cursor: pointer`

Twixes commented:
Actually, given our buttons and switches all have pointers, let's use that here too.

Twixes committed:
Use `checkboxCounter`
